straxen/plugins/nveto_pulse_processing.py

```diff
@@ -0,0 +1,804 @@
+            current_total_fraction = area / total_area
+            while current_total_fraction >= area_decils[needed_decile]:
+                if current_total_fraction:
+                    fraction = (area_decils[needed_decile] - prev_total_fraction) / current_total_fraction
+                    pos_deciles[needed_decile] = ind + fraction
+                else:
+                    pos_deciles[needed_decile] = ind
+                needed_decile += 1
+                if needed_decile > len(area_decils) - 1:
+                    # Found last area decile we have to escape while loop.
+                    compute_deciles = False
+                    break
+            prev_total_fraction = current_total_fraction
+    if current_total_fraction == 1:
+        # See hint in strax peak_properties _index_of_fraction.
+        pos_deciles[-1] = len(data)
+    return area, pos_deciles, amp, amp_ind
+
+
+@export
+@numba.njit(cache=True, nogil=True)
+def get_fwxm(data, index_maximum, percentage=0.5):
+    """
+    Estimates the left and right edge of a specific height percentage.
+
+    The function searches for the last sample below and above the specified
+    height level on the left and right hand side of the maximum. If the
+    samples are found the width is estimated based upon a linear interpolation
+    between the samples on the left and right side.
+    In case the samples cannot be found for either the right or left hand side
+    the corresponding outer bin edges are use: left 0; right last sample + 1.
+
+    Args:
+        data (np.array): Data of the pulse.
+        index_maximum (ind): Position of the maximum.
+
+    Keyword Args:
+        percentage (float): Level for which the witdth shall be computed.
+
+    Returns:
+        float: left edge [sample]
+        float: right edge [sample]
+
+    """
+    #TODO: In case of a single sample hit FWHM is not computed correctly, m becomes zero.
+    #TODO: Case for which all values are the same is not covered yet.
+
+    max_val = data[index_maximum]
+    max_val = max_val * percentage
+
+    pre_max = data[:index_maximum]
+    post_max = data[1 + index_maximum:]
+
+    # First the left edge:
+    lbi, lbs = _get_fwxm_boundary(pre_max, max_val)  # coming from the left
+    if lbi == -42:
+        # We have not found any sample below:
+        left_edge = 0.
+    else:
+        # We found a sample below so lets compute
+        # the left edge:
+        m = data[lbi + 1] - lbs  # divided by 1 sample
+        left_edge = lbi + (max_val - lbs) / m
+
+        # Now the right edge:
+    rbi, rbs = _get_fwxm_boundary(post_max[::-1], max_val)  # coming from the right
+    if rbi == -42:
+        right_edge = len(data)
+    else:
+        rbi = len(data) - rbi
+        m = data[rbi - 2] - rbs
+        right_edge = rbi - (max_val - data[rbi - 1]) / m
+
+    return left_edge, right_edge
+
+@export
+@numba.njit(cache=True, nogil=True)
+def _get_fwxm_boundary(data, max_val):
```

Comment:
To be honest I have not thought about that option :D I will check if it has any impact on the performance.

Reply:
I checked now with some mock waveform and in average using

```
%%timeit
index, value = _get_fwxm_boundary(tdata[:ind_max], hm)
index2, value2 =_get_fwxm_boundary(tdata[ind_max+1:][::-1], hm)
```

instead of 

```
%%timeit
above = np.argwhere(tdata >= hm)
index_np = above[0,0] - 1
index2_np = above[-1,0] + 1 
value_np = tdata[index_np]
value2_np = tdata[index2_np]
```is about 5 times faster.  Since I will apply this function to every single PMT signal I will keep my version.